cookbooks/cdo-varnish/test/shared/shared.rb

```diff
@@ -462,6 +462,31 @@ def does_not_strip_cookies_from_png_in_path(path)
         response = proxy_request url, {}, {}
         assert_equal 'Hello World!', last_line(response)
       end
+
+      it 'supports range requests' do
+        skip 'Development does not support range requests yet' unless environment == 'integration'
+
+        # Loop a few times since this test can fail intermittently.
+        5.times do
+          url = build_url 'x', 'sound.mp3'
+          random_string = Array.new(100_000) {[*'0'..'9', *'a'..'z', *'A'..'Z'].sample}.join
+          # Note that content-length is required
+          mock_response url, random_string, {}, {'Content-Type' => 'audio/mpeg', 'Content-Length' => '100000'}
+
+          response = proxy_request url, {'Range' => 'bytes=0-499'}
+          range = get_header(response, 'Content-Range')
+          assert_miss response
+          assert_equal 206, code(response), "Invalid response: #{response}"
+          assert_equal 'bytes 0-499/100000', range && range.chomp
+          assert_equal random_string[0..499], last_line(response)
+
+          response = proxy_request url, {'Range' => 'bytes=0-'}
+          assert_hit response
+          assert_equal 206, code(response), "Invalid response: #{response}"
+          range = get_header(response, 'Content-Range')
+          assert_equal 'bytes 0-99999/100000', range && range.chomp
+        end
+      end
```

Comment:
These tests are run manually against Varnish+CloudFront, not in CI. I had originally written this integration-test in Nov 2015, and used this test to verify that this code change fixed this issue back then.